### PR TITLE
Dd/equity calc

### PR DESCRIPTION
**Pull request by dafyddd**
Useful for calculating equity without replicating all this code.

### CODE REVIEW

File: program/src/state.rs [hunk 1]
```diff
@@ -83,12 +83,13 @@ pub enum DataType {
...
 }
 
-const NUM_HEALTHS: usize = 2;
+const NUM_HEALTHS: usize = 3;
 #[repr(usize)]
 #[derive(Copy, Clone, IntoPrimitive, TryFromPrimitive)]
 pub enum HealthType {
     Maint,
     Init,
+    Equity,
```
Comment (microwavedcola1) [inline program/src/state.rs:92]: Could we please add one line descriptions to all these variants?

Comment (microwavedcola1) [reply program/src/state.rs:92]: Still not sure, what equity means in our context 😓

File: program/src/state.rs [hunk 2]
```diff
@@ -778,7 +780,7 @@ pub struct HealthCache {
     quote: I80F48,
 
     /// This will be zero until update_health is called for the first time
-    health: [Option<I80F48>; 2],
```
Comment (ckamm) [inline program/src/state.rs:781]: use the NUM_HEALTHS constant here

File: program/src/state.rs (hunk 1 above)
Comment (ckamm) [reply program/src/state.rs:92]: Looks like it's the USD value of assets - liabilities. (all weights are 1)

File: program/src/processor.rs [hunk 3]
```diff
@@ -6652,7 +6652,7 @@ fn invoke_transfer<'a>(
 }
 
 #[inline(never)]
-fn read_oracle(
+pub fn read_oracle(
```
Comment (ckamm) [inline program/src/processor.rs:6655]: unrelated/accident?

**Review by ckamm — APPROVED** [review]

File: program/src/state.rs (hunk 2 above)
Comment (dafyddd) [reply program/src/state.rs:781]: good catch

File: program/src/processor.rs (hunk 3 above)
Comment (dafyddd) [reply program/src/processor.rs:6655]: We were using it in our onchain trading program which needed to read the oracle. It's unrelated to equity calc, but related to other programs importing mango lib

**Pull request merged**